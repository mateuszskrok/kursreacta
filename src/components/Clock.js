import React from "react";


function zFill(time,n) {
    time = time.toString()
 
    while (time.length < n) {
        time = "0" + time
    }
    return time
}

function NormalizeTime(time,min,max){
    if (time < min) {
        return min
    }
    if (time > max) {
        return max
    }
    else { return time }
}

function Clock({className = "", hours = 1, minutes = 7, seconds = 99, miliseconds =4532}) {

    
    hours = NormalizeTime(hours,0,23)
    minutes = NormalizeTime(minutes,0,59)
    seconds = NormalizeTime(seconds,0,59)
    miliseconds = NormalizeTime(miliseconds,0,999)

    hours = zFill(hours,2)
    minutes =zFill(minutes,2)
    seconds =zFill(seconds,2)
    miliseconds =zFill(miliseconds,3)

    return <h2 className={"clock" + className}>Pozosta≈Ço 
    <span className="clock__minutes"> {minutes}</span>
    <span className="clock__separator">:</span>
    <span className="clock__seconds">{seconds}</span>
    </h2>
}

export default Clock;