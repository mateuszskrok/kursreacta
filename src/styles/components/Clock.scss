.clock{
    color: orangered;

    &__minutes{
        color: blueviolet;
    }
    &__separator{
        color: chartreuse;
    }
    &__seconds{
        color: darkblue;
    }
}

