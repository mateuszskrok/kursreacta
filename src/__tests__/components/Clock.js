import React from "react";
import Clock from "../../components/Clock";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

var root = null;
var clockRenderer = null;

describe("<Clock />", () => {
    describe("when given minutes and seconds (DOM)",()=>{
        beforeEach(() => {
            root = document.createElement("div");
            ReactDOM.render(
             <Clock minutes={10} seconds={20} />, root
            )
        });
        it('renders with class clock', () => {
           expect(root.childNodes[0].className).toMatch(/clock/)
        });
    
        it('renders as h2', () => {
            expect(root.childNodes[0].nodeName).toEqual("H2")
         });

        it('renders time correctly', () => {
            expect(<Clock minutes={10} seconds={20}/>).toEqual(<Clock className="" hours={0} miliseconds={0} minutes={10} 
            seconds={20} />)
        });
    
    })
    describe("when given minutes and seconds (test renderer)",()=>{
        beforeEach(() => {
            clockRenderer = renderer.create(
                <Clock minutes={10} seconds={20} />
            );
        });
        it('renders correctly', () => {
            expect(clockRenderer.toJSON()).toMatchSnapshot();
        });
        it('renders with className clock', () => {
            expect(clockRenderer.toJSON().props).toMatchObject({"className": expect.stringMatching(/clock/)})
        });
    
        it('renders as h2', () => {
            expect(clockRenderer.toJSON().type).toEqual("h2")
            
         });

        it('renders time corectly', () => {
            expect(clockRenderer.toJSON().children).toEqual(expect.arrayContaining(["10","20"]))
        });
    
    })
   
});