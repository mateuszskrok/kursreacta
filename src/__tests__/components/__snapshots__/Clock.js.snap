// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> when given minutes and seconds (test renderer) renders correctly 1`] = `
<h2
  className="clock"
>
  Pozosta≈Ço
  10
  :
  20
</h2>
`;
